pkg/tasks/repository_snapshot.go

```diff
@@ -284,6 +296,54 @@ func (sr *SnapshotRepository) lookupRepoObjects() (api.RepositoryResponse, error
 	return repoConfig, nil
 }
 
+func (sr *SnapshotRepository) cleanupOnCancel() error {
+	logger := LogForTask(sr.task.Id.String(), sr.task.Typename, sr.task.RequestID)
+	// TODO In Go 1.21 we could use context.WithoutCancel() to make copy of parent ctx that isn't canceled
+	ctxWithLogger := logger.WithContext(context.Background())
+	pulpClient := pulp_client.GetPulpClientWithDomain(ctxWithLogger, sr.domainName)
+	if sr.payload.SyncTaskHref != nil {
+		task, err := pulpClient.CancelTask(*sr.payload.SyncTaskHref)
+		if err != nil {
+			return err
+		}
+		task, err = pulpClient.GetTask(*sr.payload.SyncTaskHref)
+		if err != nil {
+			return err
+		}
+		versionHref := pulp_client.SelectVersionHref(&task)
+		if versionHref != nil {
+			_, err = pulpClient.DeleteRpmRepositoryVersion(*versionHref)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	if sr.payload.DistributionTaskHref != nil {
+		task, err := pulpClient.CancelTask(*sr.payload.DistributionTaskHref)
+		if err != nil {
+			return err
+		}
+		task, err = pulpClient.GetTask(*sr.payload.DistributionTaskHref)
+		if err != nil {
+			return err
+		}
+		versionHref := pulp_client.SelectRpmDistributionHref(&task)
+		if versionHref != nil {
+			_, err = pulpClient.DeleteRpmDistribution(*versionHref)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	if sr.snapshotUUID != "" {
+		err := sr.daoReg.Snapshot.Delete(sr.snapshotUUID)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
I don't see a publication task cancel.  I don't think publication needs to be explicitly deleted, but i'm not sure what happens if you delete a version while its in the middle of being published?

Reply:
good point. I'll add that in.